### PR TITLE
Improve psalm types for `DatabaseInterface::transaction()`

### PR DESCRIPTION
**Pull request by vjik**
## What was changed

Add more specific type for "DatabaseInterface::transaction()".

## Why?

Psalm error in app. I am waiting callback result as method result.

## Checklist

- How was this tested: psalm (count errors same before and after)
  
  
- Any docs updates needed? No

### CODE REVIEW

File: src/DatabaseInterface.php [hunk 1]
```diff
@@ -135,6 +135,10 @@ public function select(mixed $columns = '*'): SelectQuery;
      * @link http://en.wikipedia.org/wiki/Database_transaction
      *
      * @throws \Throwable
+     *
+     * @psalm-template CallbackResult
+     * @psalm-param callable():CallbackResult $callback
+     * @psalm-return CallbackResult
```
Comment (roxblnfk) [inline src/DatabaseInterface.php]:
```suggestion
     * @template CallbackResult
     *
     * @param callable(): CallbackResult $callback
     *
     * @preturn CallbackResult
```

**Review by roxblnfk — CHANGES_REQUESTED** [review]
Please fill the form in the PR description

**Comment by vjik** [comment]
> Please fill the form in the PR description

Done

**Review by roxblnfk — APPROVED** [review]

**Comment by roxblnfk** [comment]
Thanks

**Pull request merged**